cura/Settings/ExtruderStack.py

```diff
@@ -119,6 +51,33 @@ def getNextStack(self) -> Optional["GlobalStack"]:
     def getLoadingPriority(cls) -> int:
         return 3
 
+    ##  Return the filament diameter that the machine requires.
+    #
+    #   If the machine has no requirement for the diameter, -1 is returned.
+    #   \return The filament diameter for the printer
+    @property
+    def materialDiameter(self) -> float:
+        containers_to_check = [self.variant, self.definitionChanges, self.definition]
```

Comment:
Instead of checking these three, we should probably ask the stack and skip until after the material (or just bloody split the setting up into the required diameter and the actual diameter, but that's for another time I suppose).

We're bound to add more containers to the stack later. And if we do, it'll be below the material profile. We'd forget about updating it here as well.

Reply:
Changed in a2c2282b4